src/fmelt.c

```diff
@@ -478,29 +478,19 @@ SEXP getvaluecols(SEXP DT, SEXP dtnames, Rboolean valfactor, Rboolean verbose, s
         }
       }
         break;
-      case INTSXP : {
+      case INTSXP :
+      case LGLSXP : {
         int *itarget = INTEGER(target);
         const int *ithiscol = INTEGER(thiscol);
         if (data->narm) {
-          for (k=0; k<thislen; k++)
-            itarget[counter + k] = ithiscol[ithisidx[k]-1];
-        } else {
-          memcpy(itarget + j*data->nrow, ithiscol, data->nrow*size);
-        }
-      }
-        break;
-      case LGLSXP : {
-        int *itarget = LOGICAL(target);
-        const int *ithiscol = LOGICAL(thiscol);
-        if (data->narm) {
-          for (k=0; k<thislen; k++)
+          for (int k=0; k<thislen; ++k)
             itarget[counter + k] = ithiscol[ithisidx[k]-1];
         } else {
           memcpy(itarget + j*data->nrow, ithiscol, data->nrow*size);
         }
-      }
-        break;
-      default : error("Unknown column type '%s' for column '%s'.", type2char(TYPEOF(thiscol)), CHAR(STRING_ELT(dtnames, INTEGER(thisvaluecols)[i]-1)));
+      } break;
+      default :
```

Comment:
I would keep it in same line, so if you collapse `{` of switch then every case, including default, takes only single line

Reply:
I like long lines as you know, but this seems a step too far to put the error on the same line as the default:  For the reason that the other cases in this switch aren't like that due to being bigger.  When all the cases are one short line, then yes but not in this case. Consistency within the switch.

Reply:
then what can help here is
```
default : {
  error("a");
}
```
collapse-able to single line